### ISSUE TITLE
Oriented Bounding Box returns reflection, not rotation

### ISSUE THREAD

**Issue by richardrl** [isl-org/Open3D#4006]
```
    obb = open3d.geometry.OrientedBoundingBox()
    obb = obb.create_from_points(open3d.utility.Vector3dVector(rotated_pointcloud.cpu().data.numpy()))
    original_rotation = copy.deepcopy(obb.R)

  original_rotation
  array([[ 0.61638224, -0.78729557,  0.01544744],
         [-0.787433  , -0.61613522,  0.01807361],
         [ 0.00471156,  0.02330407,  0.99971732]])
  
(Pdb) np.linalg.det(original_rotation)
-1.0

```

This should not happen, right?

**Comment by mchuruvija** [comment]
Hey, I'm experiencing that same issue: sometimes I get pure rotation and sometimes reflection.

`PfLEO_OBB = o3d.geometry.OrientedBoundingBox.create_from_points(PfLEO.points)
r11, r12, r13, r21, r22, r23, r31, r32, r33 = PfLEO_OBB.R[0][0], PfLEO_OBB.R[0][1], PfLEO_OBB.R[0][2], PfLEO_OBB.R[1][0], PfLEO_OBB.R[1][1], PfLEO_OBB.R[1][2], PfLEO_OBB.R[2][0], PfLEO_OBB.R[2][1], PfLEO_OBB.R[2][2]
dx, dy, dz = PfLEO_OBB.center[0], PfLEO_OBB.center[1], PfLEO_OBB.center[2]
MfLEOtPCk = ht.Transformation(matrix=[
        [r11, r12, r13, dx],
        [r21, r22, r23, dy],
        [r31, r32, r33, dz],
        [0., 0., 0., 1.]])
vis.add_geometry(o3d.geometry.TriangleMesh.create_coordinate_frame(size=0.05, origin=[0, 0, 0]).transform(MfLEOtPCk.getTransformation()), reset_bounding_box=False)
`
![image](https://user-images.githubusercontent.com/57736542/132401040-01366f66-fe1f-4f08-a1cf-b32559e46741.png)

Is there a workaround that?

**Comment by melsanharawi** [comment]
import numpy as np

reflect = np.asarray([[-1], [-1.], [-1.] ])

def getRotationMatrix(s):
    finalRot = np.linalg.inv(s.get_oriented_bounding_box().R)
    det = np.linalg.det(s.get_oriented_bounding_box().R)
    if det < 0:
        finalRot = np.multiply(finalRot, reflect)
    return finalRot

**Comment by melsanharawi** [comment]
just a workaround :)

**Comment by benjaminum** [comment]
Thanks for reporting. Fixed in #4452
